subsys/net/lib/lwm2m/Kconfig

```diff
@@ -38,6 +38,21 @@ config LWM2M_SERVER_OBJECT_VERSION_1_1
 
 endchoice
 
+choice
+	prompt "LwM2M Connectivity Monitor object version"
+	default LWM2M_CONNMON_OBJECT_VERSION_1_0
```

Comment:
`depends on LWM2M_CONN_MON_OBJ_SUPPORT`

And I'd move this somewhere around line 259, where the Connectivity monitor object is actually enabled. Perhaps at some point we'd need to clean up a bit, specifically move all core objects configurations to a separate file, similarry to IPSO objects.